extension/docs/WebComponentsIntegrationGuide.md

```diff
@@ -0,0 +1,665 @@
+```json
+{
+  "pspReference": "853592567856061C",
+  "resultCode": "Authorised",
+  "amount": {
+    "currency": "EUR",
+    "value": 1000
+  },
+  "merchantReference": "YOUR_ORDER_REFERENCE"
+}
+```
+A commercetools payment with `submitAdditionalPaymentDetailsResponse` field with the response above.
+Notice that a transaction is added to the payment. The transaction is of type `Authorization` 
+and has `amount` taken from `amountPlanned`. `interactionId` is matching the `makePaymentResponse` 
+```json
+{
+  "amountPlanned": {
+    "type": "centPrecision",
+    "currencyCode": "EUR",
+    "centAmount": 1000,
+    "fractionDigits": 2
+  },
+  "custom": {
+    "type": {
+      "typeId": "type",
+      "key": "ctp-adyen-integration-web-components-payment-type"
+    },
+    "fields": {
+      "submitPaymentDetailsRequest": "{\"details\":{\"redirectResult\":\"Ab02b4c0!...\"}}",
+      "submitAdditionalPaymentDetailsResponse": "{\"pspReference\":\"853592567856061C\",\"resultCode\":\"Authorised\",\"amount\":{\"currency\":\"EUR\",\"value\":1000},\"merchantReference\":\"YOUR_ORDER_REFERENCE\"}"
+    }
+  },
+  "transactions": [
+    {
+      "id": "eab650fd-8616-471b-b884-eef641b4f169",
+      "type": "Authorization",
+      "amount": {
+        "type": "centPrecision",
+        "currencyCode": "EUR",
+        "centAmount": 1000,
+        "fractionDigits": 2
+      },
+      "interactionId": "853592567856061C",
+      "state": "Success"
+    }
+  ]
+}
+```
+
+## Step 6: Capture payment (required for Klarna)
+All Klarna payments [have to be manually captured](https://docs.adyen.com/payment-methods/klarna/web-component#capture) within 28 days after authorisation, even if you have enabled automatic capture on your merchant account.
+Refer to [Manual Capture](ManualCapture.md) guide to see how it can done.
+
+## Error handling
+
+In case you encounter errors in your integration, refer to the following:
+
+### Extension module errors
+
+If you receive a `non-HTTP 200 response`, use the commercetools `interface interactions` to troubleshoot the request and errors.
+
+Interface interactions can be requests sent to the Adyen, responses received from the Adyen or notifications received from the Adyen. 
+Some interactions may result in a transaction. If so, the interactionId in the Transaction should be set to match the `pspReference` of the Adyen for the interaction.
+
+### Adyen payment refusals
+
+If you receive an HTTP 200 response with an Error or Refused resultCode, check the refusal reason and, if possible, modify your request.
+
+Check the following table to see the mapping of Adyen [result codes](https://docs.adyen.com/development-resources/response-handling#error-codes-types) to commercetools [transaction state](https://docs.commercetools.com/http-api-projects-payments#transactionstate)
+|Adyen result code| The commercetools transaction state
+| --- | --- |
+| Authorised| Success|
+| Refused| Failure|
+| Error| Failure|
```

Comment:
In case of Error or Refused do we create a transaction at all? If not, it means that failure will not be set to the transaction?

Reply:
We are creating Authorization transaction with those states.

https://github.com/commercetools/commercetools-adyen-integration/blob/eb945e9e1fd373b2078e05cefe19573e7d0601c7/extension/src/paymentHandler/payment-utils.js#L96-L105

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/329/commits/f95f269c2df4190109589c26117e8937312afa21